api/bootstrap/kubeadm/v1beta2/kubeadm_types.go

```diff
@@ -180,6 +180,20 @@ type ClusterConfiguration struct {
 	// featureGates enabled by the user.
 	// +optional
 	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+
+	// certificateValidityPeriodDays specifies the validity period for a non-CA certificate generated by kubeadm.
+	// If not specified, kubeadm will use a default of 3650 days (10 years).
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=10950
+	CertificateValidityPeriodDays int32 `json:"certificateValidityPeriodDays,omitempty"`
+
+	// caCertificateValidityPeriodDays specifies the validity period for a CA certificate generated by kubeadm.
+	// If not specified, kubeadm will use a default of 3650 days (10 years).
```

Comment:
```suggestion
	// If not specified, Cluster API will use a default of 3650 days (10 years).
	// This field cannot be modified.
```
(It is us in this case, not kubeadm)